### PR TITLE
Use `setup-universal-sierra-compiler` action

### LINKED ISSUE #1858: Add and integrate an action for USC for `setup-snfoundry`
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

Please try using composite actions in order to achieve that

When closing - please unblock https://github.com/foundry-rs/starknet-foundry/issues/1834

### PR DESCRIPTION
**Pull request by tomek0123456789**
Closes #1858

## Introduced changes

- Added [`setup-universal-sierra-compiler`](https://github.com/software-mansion/setup-universal-sierra-compiler) action

Couldn't integrate it using composite actions, sadly.

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by piotmag769 — APPROVED** [review]

File: .github/workflows/release.yml
Comment (THenry14) [inline .github/workflows/release.yml:193]: From what I've seen, a support for windows has been added to the gh action - that is cool. Do you happen to know if there any plans to also include this support to install.sh script?

File: .github/workflows/release.yml
Comment (THenry14) [inline .github/workflows/release.yml:196]: I can't comment on the lines below, but there we are using `UDC_PATH`, that is used in this Smoke tests job step - I think it should be removed, otherwise it may break smoke tests (UDC_PATH overwrites UNIVERSAL_SIERRA_COMPILER, and after this is merged I think it will set it to an empty string. This variable is being used by snforge/sncast if I recall correctly, so there will be no path to usc)

**Review by THenry14 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml
Comment (tomek0123456789) [reply .github/workflows/release.yml:193]: Sure, I could do that later

File: .github/workflows/release.yml
Comment (tomek0123456789) [reply .github/workflows/release.yml:196]: Done!

**Review by Arcticae — APPROVED** [review]

**Review by THenry14 — APPROVED** [review]

**Comment by tomek0123456789** [comment]
Will merge after making sure that release action works as intended, right now it doesn't success on my fork.

**Comment by THenry14** [comment]
> Will merge after making sure that release action works as intended, right now it doesn't success on my fork.

There may be issues when running on a fork. I suppose you could comment out the logic that deploys docs + creates a release, and set the triggering branch to your own. That way, the release flow (building, testing) will be properly tested on our env, just without the last step (release itself). Makes sense?

**Comment by tomek0123456789** [comment]
Release check on fork ran here: https://github.com/tomek0123456789/starknet-foundry/actions/runs/8554397339/job/23481990666

**Pull request merged**
